src/bluesearch/entrypoint/database/run.py

```diff
@@ -0,0 +1,636 @@
+        for uid in all_uids:
+            query = sqlalchemy.text(
+                "SELECT article_id from articles WHERE article_id = :uid"
+            )
+            res = engine.execute(query, uid=uid).fetchall()
+
+            if not res:
+                new_uids.append(uid)
+
+        return not new_uids
+
+    def program_args(self) -> list[str]:
+        """Define subprocess arguments."""
+        input_dir = Path(self.input().path)
+
+        command = [
+            *BBS_BINARY,
+            "add",
+            *VERBOSITY,
+            self.db_url,
+            input_dir,
+            f"--db-type={self.db_type}",
+        ]
+
+        return command
+
+
+def get_all_dependencies(task: luigi.Task) -> set[luigi.Task]:
+    """Get all dependencies of a given task.
+
+    Parameters
+    ----------
+    task
+        Input task
+
+    Returns
+    -------
+    set[luigi.Task]
+        All the tasks that the `input` depends on including itself.
+    """
+    current_deps = set(task.deps())
+    if not current_deps:
+        return set()
+
+    else:
+        deps = {task}
+        for current_dep in current_deps:
+            deps |= get_all_dependencies(current_dep)
+
+        return deps | current_deps
+
+
+def run(
+    *,
+    source: str,
+    from_month: str,
+    filter_config: Path,
+    output_dir: Path,
+    db_url: str,
+    db_type: str,
+    mesh_topic_db: Path | None,
+    dry_run: bool,
+    grobid_host: str | None,
+    grobid_port: int | None,
+    identifier: str | None,
+    final_task: str | None,
+) -> int:
+    """Run overall pipeline.
+
+    Parameter description and potential defaults are documented inside of the
+    `get_parser` function.
+    """
+    logger.info("Starting the overall pipeline")
+
+    DownloadTask.capture_output = CAPTURE_OUTPUT
+    UnzipTask.capture_output = CAPTURE_OUTPUT
+    TopicExtractTask.capture_output = CAPTURE_OUTPUT
+    TopicFilterTask.capture_output = CAPTURE_OUTPUT
+    PerformFilteringTask.capture_output = CAPTURE_OUTPUT
+    ConvertPDFTask.capture_output = CAPTURE_OUTPUT
+    ParseTask.capture_output = CAPTURE_OUTPUT
+    AddTask.capture_output = CAPTURE_OUTPUT
+
+    add_task_inst = AddTask(
+        source=source,
+        from_month=from_month,
+        filter_config=str(filter_config),
+        output_dir=str(output_dir),
+        mesh_topic_db=str(mesh_topic_db),
+        grobid_host=grobid_host,
+        grobid_port=grobid_port,
+        db_url=db_url,
+        db_type=db_type,
+        identifier=identifier,
+    )
+    if final_task is None:
+        selected_task_inst = add_task_inst
+    else:
+        all_dependencies = get_all_dependencies(add_task_inst)
+        all_dependencies_map = {t.__class__.__name__: t for t in all_dependencies}
+
+        if final_task in all_dependencies_map:
+            selected_task_inst = all_dependencies_map[final_task]
+        else:
+            raise ValueError(f"Unrecognized final task {final_task}")
+
+    luigi_kwargs = {
```

Comment:
An alternative could be to have those `luigi_conf` as default, but allow user to overwrite them on the CLI by passing a list of key-value pairs that will be passed to luigi, like this:
```bash
bbs_database run ... --luigi-config=log_level:WARNING,local_scheduler:True,...
```
Especially because otherwise changing things like the scheduler (local/central) would require modifying every time the source code, which IMO is not ideal...

Reply:
I like the idea;)

Again, this PR was just a sketch and all the luigi configuration was hardcoded to whatever was the most convenient when developing. IMO we could even do this in a separate PR since it might be worth looking into what the best practices are when it comes to providing "dictionary-like" data via CLI.

Reply:
I tested now, and the configuration can indeed be moved away from the source code.
Just modify the `luigi.cfg` to look as follows:
```python
[core]
    autoload_range = True
    log_level = DEBUG
    local_scheduler = True
```

Reply:
Done in `8830e69`. I set the logging level to `INFO` since it seems to be a reasonable default.